src/h3lib/lib/latLng.c

```diff
@@ -257,23 +257,28 @@ void _geoAzDistanceRads(const LatLng *p1, double az, double distance,
  * handled with hardwired static values, but it may be worthwhile to put these
  * static values into another file that can be autogenerated by source code in
  * the future.
+ * AJ TODO 2021-10-30: update this comment with the python notebooks when landed
  */
 
 double H3_EXPORT(getHexagonAreaAvgKm2)(int res) {
     static const double areas[] = {
-        4250546.848, 607220.9782, 86745.85403, 12392.26486,
-        1770.323552, 252.9033645, 36.1290521,  5.1612932,
-        0.7373276,   0.1053325,   0.0150475,   0.0021496,
-        0.0003071,   0.0000439,   0.0000063,   0.0000009};
+        4.357449416078383e+06, 6.097884417941332e+05, 8.680178039899720e+04,
```

Comment:
I'm a bit uncomfortable with the approach in the notebook - I believe when @sahrk calculated the original values, he used the known surface area of the Earth, rather than calculating it based on the cells. Your notebook is internally consistent - you should be able to get the average area from the individual cell areas *as calculated in the library* for each res - but it's clearly off by several percentage points from the "true" area of the cells as calculated on a non-spherical model.

I'm not sure where this leaves us, because I don't have a good opinion about which is better, but I think we should definitely have a clearly explained reason to pick one or the other in the docs.

Reply:
The known surface area of the earth is equivalent to the sum of the areas of the res 0 cells. We have a test that verifies it: https://github.com/uber/h3/blob/32b6e07f41353e9def5a778fc5a4a7c8cdddd93a/src/apps/testapps/testH3CellAreaExhaustive.c#L156-L185

I was just using sum of the areas as a quick way to get the total earth area, since we don't expose the authalic radius to the user (I don't think...).

@nrabinowitz: But it sounds like you have two concerns. Does the above alleviate your concerns about the first?

Reply:
OK, I think this is fine - I don't think I realized that the spherical model with the authalic radius would yield the same surface area as the geoid model, which I guess is what "authalic" means 🤦 . I thought the discrepancies between the original table and the new table were due to differences in the base Earth area being used, but now I'm guessing they were due to the way @sahrk handled pentagons.

So... nvm, I guess? I do think we should have a bit more gloss on the docs page to make it very clear what's being calculated, given how many questions we get on this.

Reply:
I added a note about the authalic radius.

I repeat it a few times, hopefully making it hard to miss, but maybe it's too much?